cms/static/js/spec/views/active_video_upload_list_spec.js

```diff
@@ -51,41 +72,235 @@ define(
+                    responseText: JSON.stringify({
+                        files: _.map(
+                            fileNames,
+                            function(fileName) {
+                                return {
+                                    edx_video_id: VIDEO_ID,
+                                    file_name: fileName,
+                                    upload_url: url || makeUploadUrl(fileName)
+                                };
+                            }
+                        )
+                    })
+                });
+            };
+
+            describe('errors', function() {
+                it('should show error notification for status update request in case of server error', function() {
+                    this.view.sendStatusUpdate([
+                        {
+                            edxVideoId: '101',
+                            status: 'upload_completed',
+                            message: 'Uploaded completed'
+                        }
+                    ]);
+                    getStatusUpdateRequest().respondWith(
+                        {
+                            status: 500,
+                            responseText: JSON.stringify({
+                                error: '500 server errror'
+                            })
+                        }
+                    );
+                    expect($('#notification-error-title').text().trim()).toEqual(
+                        "Studio's having trouble saving your work"
+                    );
+                    expect($('#notification-error-description').text().trim()).toEqual('500 server errror');
+                });
+
+                it('should correctly parse and show S3 error response xml', function() {
+                    var fileInfo = {name: 'video.mp4', size: 10000},
+                        videos = {
+                            files: [
+                                fileInfo
+                            ]
+                        },
+                        S3Url = 'http://s3.aws.com/upload/videos/' + fileInfo.name,
+                        requests;
+
+                    // this is required so that we can use AjaxHelpers ajax mock utils instead of jamin mock-ajax.js
+                    jasmine.Ajax.uninstall();
+
+                    requests = AjaxHelpers.requests(this);
+
+                    this.view.$uploadForm.fileupload('add', videos);
+                    AjaxHelpers.respond(requests, {
+                        status: 200,
+                        body: {
+                            files: [{
+                                edx_video_id: VIDEO_ID,
+                                file_name: fileInfo.name,
+                                upload_url: S3Url
+                            }]
+                        }
+                    });
+                    expect(requests.length).toEqual(2);
+                    AjaxHelpers.respond(
+                        requests,
```

Comment:
Should we only check the S3 request, that would be second request in this case ?